Create More Faces CheckPoint Tests

volosied opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

for https://github.com/OpenLiberty/open-liberty/issues/31286

volosied commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=a50f6bc0-8b70-4fd9-a9fb-2e207807c046) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

tjwatson commented on dev/io.openliberty.checkpoint_fat_faces/bnd.bnd:
```diff
@@ -0,0 +1,69 @@
+    servlet-5.0,\
+    servlet-6.0,\
+    servlet-6.1,\
+    validation-3.1
```
you need to add `checkpoint` to the list of `tested.features` so the test will run in the checkpoint run of the build.

tjwatson replied:
nevermind, `checkpoint` is there already.

volosied commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=54fdf2a6-718a-424d-9343-1a8e5ef3618d) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

tjwatson reviewed (dismissed):
LGTM

I assume the various application sources are a copy from another FAT and that is the reason the copyright date hasn't been updated on them.

pnicolucci commented on dev/io.openliberty.checkpoint_fat_faces/bnd.bnd:
```diff
@@ -0,0 +1,69 @@
+
+
+-buildpath: \
+	com.ibm.websphere.javaee.annotation.1.2;version=latest,\
```
Can you clean up this formatting in the bnd?

pnicolucci commented on dev/io.openliberty.checkpoint_fat_faces/fat/src/io/openliberty/checkpoint/fat/FATSuite.java:
```diff
@@ -0,0 +1,62 @@
+
+@RunWith(Suite.class)
+@SuiteClasses({
+                AlwaysPassesTest.class,
```
Why do we need the AlwaysPassesTest?

volosied replied:
If this bucket doesn't run on a CRIU-capable machine, then the bucket should still pass. We don't want false positives.  CRIU is Linux specific.

pnicolucci commented on dev/io.openliberty.checkpoint_fat_faces/publish/servers/jsf23CDIGeneralServer/bootstrap.properties:
```diff
@@ -0,0 +1,16 @@
+###############################################################################
+bootstrap.include=../testports.properties
+osgi.console=7777
+com.ibm.ws.logging.trace.specification=*=info=enabled:com.ibm.ws.app.manager.*=all:org.apache.myfaces.*=all:com.ibm.ws.jsf*=all:com.ibm.ws.runtime.update.*=all:logservice=detail
```
do we need all this tracing enabled? Same comment for the other server

pnicolucci commented on dev/io.openliberty.checkpoint_fat_faces/.project:
```diff
@@ -0,0 +1,34 @@
+			<name></name>
+			<type>30</type>
+			<matcher>
+				<id>org.eclipse.core.resources.regexFilterMatcher</id>
```
Is this necessary?

pnicolucci reviewed (commented):
I don't love the idea of copying all of these test applications to a new FAT when they already exist in the JSF 2.3 FATs. Is there a way we can just add checkpoint test coverage to our existing FATS? Would it add a lot of execution time and that is why we're doing a new FAT bucket? This is a good test coverage addition, thanks for working on this!

Commit:
Enable CheckPoint in n JSFSimpleHtmlUnit

Commit:
Enable Checkpoint in JSF22ComponentRendererTests

Check in other JSF 22 FAT files

Fix up JSF 2.2 FAT

Commit:
Enable CheckPoint in JSF23UIRepeatConditionTests

Update JSF 2.3 FAT

Move improt in JSF23UIRepeatConditionTests

Add comment to FATSuite

volosied commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=39504896-067a-4d7a-8488-feb227470795) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

volosied committed:
Skip Checkpoint for EE7 Runs

volosied commented:
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=987012db-715e-4649-a625-534efcd17074) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

pnicolucci commented on dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22ComponentRendererTests.java:
```diff
@@ -12,6 +12,7 @@
 import static org.junit.Assert.assertTrue;
```
Could use a date range copy right update to include 2025

volosied committed:
Fix copyright and remove wrongly added annotation

volosied commented:
!libby ([view Open Liberty Delivery Requirements Verification](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=db3ba517-fc71-47be-bc61-74e88d871f0b) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._

tjwatson commented on dev/com.ibm.ws.jsf.2.2_fat/fat/src/com/ibm/ws/jsf22/fat/tests/JSF22ComponentTesterTests.java:
```diff
@@ -51,6 +52,7 @@
  * Tests to execute on the jsfTestServer2 that use HtmlUnit.
  */
 @Mode(TestMode.FULL)
+@CheckpointTest(alwaysRun = true)
```
I think this is not necessary?  Is this a checkpoint tests?

volosied replied:
I completely overlooked this. Thanks for spotting it.

tjwatson reviewed (commented):
LGTM

Just one nit comment.  for a `@CheckpointTest(alwaysRun = true)` that I think is not needed.

volosied commented:
!libby ([view Open Liberty Delivery Requirements Verification](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=2a1519cf-7f64-4aa9-bd48-a95326421c78) - :white_check_mark: completed successfully!)

_Note: Target locations of links might be accessible only to IBM employees._
